x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -217,6 +219,14 @@ public static Boolean isTokenServiceEnabled(Settings settings) {
         return XPackSettings.TOKEN_SERVICE_ENABLED_SETTING.get(settings);
     }
 
+    private SecurityIndexManager indexManager() {
+        if (securityTokensIndex.exists()) {
```

Comment:
shouldn't we base this on the index format? also what about tokens created/invalidated while the upgrade is running

Reply:
Alternatively, we can make sure that any write operations are retried.

Reply:
I think the read only aspect during the upgrade is definitely acceptable.